src/core/packages/saved-objects/server/src/model_version/model_change.ts

```diff
@@ -161,12 +161,16 @@ export interface SavedObjectsModelDataRemovalChange {
  *         Please reach out to the Core team if you think you need to use this, as you theoretically shouldn't.
  */
 export interface SavedObjectsModelUnsafeTransformChange<
-  PreviousAttributes = any,
-  NewAttributes = any
+  PreviousAttributes = unknown,
+  NewAttributes = unknown
 > {
   type: 'unsafe_transform';
   /**
    * The transform function to execute.
    */
-  transformFn: SavedObjectModelUnsafeTransformFn<PreviousAttributes, NewAttributes>;
+  transformFn: (
+    cast: <A, B>(
+      fn: SavedObjectModelUnsafeTransformFn<A, B>
+    ) => SavedObjectModelUnsafeTransformFn<unknown, unknown>
+  ) => SavedObjectModelUnsafeTransformFn<unknown, unknown>;
```

Comment:
The best idea I've got in which I tried to leverage just TS was this `cast` helper... The issue is we can't go from unconstrained generic to constrained generic. That's why the issue was bubbling up the type hierarchy. But we can go from constrained to unconstrained which is why this cast AB hack thingy works.